launcher/src/main/java/org/apache/spark/launcher/SparkClassCommandBuilder.java

```diff
@@ -80,12 +81,18 @@
     }
 
     List<String> cmd = buildJavaCommand(extraClassPath);
+
     for (String key : javaOptsKeys) {
-      addOptionString(cmd, System.getenv(key));
+      String envValue = System.getenv(key);
+      if (!isEmpty(envValue) && envValue.contains("Xmx")) {
```

Comment:
I think we could just do the contains on the final cmd string after the iterating.  So move this outside the for loop.